samtranslator/model/api/api_generator.py

```diff
@@ -626,14 +676,31 @@ def _construct_alias_target(self, domain: Dict[str, Any], api_domain_name: str,
         return alias_target
 
     @cw_timer(prefix="Generator", name="Api")
-    def to_cloudformation(self, redeploy_restapi_parameters, route53_record_set_groups):  # type: ignore[no-untyped-def]
+    def to_cloudformation(
+        self, redeploy_restapi_parameters: Optional[Any], route53_record_set_groups: Dict[str, Route53RecordSetGroup]
+    ) -> Tuple[
+        ApiGatewayRestApi,
+        ApiGatewayDeployment,
+        ApiGatewayStage,
+        List[LambdaPermission],
+        Optional[ApiGatewayDomainName],
+        Optional[List[ApiGatewayBasePathMapping]],
+        Optional[Route53RecordSetGroup],
+        Optional[Any],
+        Optional[List[Route53RecordSet]],
+    ]:
```

Comment:
omg this is... can we improve this function to make it return `List[SamResource]` or `List[Any]`

Reply:
Sure that's ugly.

Reply:
Maybe a dataclass? With a `List` we have to remember what resource that index matches to.

Reply:
I don't think there is flexibility. `to_cloudformation` has to return an iterable of CFN resources

Reply:
Do the specific types matter? Shouldn't they all be by design `Resource`s?

Reply:
Oh right! Yes a List of `Resource` should work

Reply:
Looking at the code it does depend on the specific value though, so IMO `Tuple` makes the most sense without large refactor. 🤷‍♂️ Your call.

Reply:
Did a little refactoring so it can return a list now.